tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,249 @@
+    in the REPORT file is also displayed in the log.
+
+    Args:
+        action_text -- Title of last action taken
+                (usually adding nodes or adding osds)
+        elastic_info -- ElasticData object
+    """
+    output_info = {"title": action_text}
+    pod_obj = ocp.OCP(
+        kind=constants.POD, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+    )
+    osd_list = pod_obj.get(selector=constants.OSD_APP_LABEL)["items"]
+    node_stats = {}
+    for osd_ent in osd_list:
+        osd_node = osd_ent["spec"]["nodeName"]
+        if osd_node in node_stats:
+            node_stats[osd_node].append(osd_ent)
+        else:
+            node_stats[osd_node] = [osd_ent]
+    osds_per_node = []
+    for entry in node_stats:
+        osds_per_node.append(len(node_stats[entry]))
+    wnodes = get_nodes(constants.WORKER_MACHINE)
+    for wnode in wnodes:
+        if wnode.name not in node_stats:
+            osds_per_node.append(0)
+    maxov = max(osds_per_node)
+    minov = min(osds_per_node)
+    this_skew = maxov - minov
+    logging.info(f"Skew found is {this_skew}")
+    output_info["osds"] = osd_list
+    output_info["worker_nodes"] = wnodes
+    output_info["pairings"] = {}
+    for entry in osd_list:
+        output_info["pairings"][entry["metadata"]["name"]] = entry["spec"]["nodeName"]
+    output_info["maxov"] = maxov
+    output_info["minov"] = minov
+    output_info["skew_value"] = this_skew
+    balanced = True
+    if this_skew > 1 and maxov > MAX_OSDS_PER_NODE:
+        balanced = False
+    elastic_info.add_key(elastic_info.record_counter, output_info)
+    elastic_info.log_recent_activity()
+    elastic_info.record_counter += 1
+    if not balanced:
+        logging.info("OSDs are not balanced")
+        if action_text == FINAL_REPORT:
+            logging.info("FINAL RESULT -- OSDs are not balanced")
+            ceph_health_check(tries=30, delay=60)
+            return
+    if action_text == FINAL_REPORT:
+        logging.info("FINAL RESULT -- OSDs are balanced")
+        ceph_health_check(tries=30, delay=60)
+
+
+class ElasticData(PerfResult):
+    """
+    Wrap PerfResult and keep track of a counter to be used as
+    an index into the table data saved.
+    """
+
+    def __init__(self, uuid, crd):
+        super(ElasticData, self).__init__(uuid, crd)
+        self.index = "test_osd_node_balancing"
+        self.new_index = "test_osd_node_balancing_new"
+        self.record_counter = 0
+
+    def log_recent_activity(self):
+        new_data = self.results[self.record_counter]
+        logging.info(new_data["title"])
+        logging.info("pairings:")
+        for entry in new_data["pairings"]:
+            logging.info(f"     {entry} -- {new_data['pairings'][entry]}")
+        logging.info(f"maxov: {new_data['maxov']}")
+        logging.info(f"minov: {new_data['minov']}")
+        logging.info(f"skew_value: {new_data['skew_value']}")
+
+
+@ignore_leftovers
+@scale
```

Comment:
@wusui we should not execute this TC with scale marker. As we are planning to add one more marker(not started any work on that yet) especially for scale TC which we can't perform teardown at the end of test.
I see the TC is already completed and with scale marker next testcase is going to fail in suite execution
1. Adding the new marker to setup and teardown the cluster after every test execution
2. Is there a way to execute this at end which is temporary solution
3. Other option would be to merge this PR with some dummy marker and later once we have the marker to teardown the cluster, we can update this code.

Reply:
@keesturam ^^